stockflux-launcher/src/free-text-search/FreeTextSearch.js

```diff
@@ -0,0 +1,216 @@
+import React, {
+  useState,
+  useReducer,
+  useEffect,
+  useRef,
+  useCallback
+} from 'react';
+import { FaSearch } from 'react-icons/fa';
+import { StockFlux } from 'stockflux-core';
+import SearchResult from './search-result';
+import {
+  InterApplicationBusHooks,
+  WindowHooks,
+  Constants
+} from 'openfin-react-hooks';
+import { createWindow, populateResultsContainer } from './SearchResultWindow';
+import './FreeTextSearch.css';
+
+const SEARCHING = 'searching';
+const SUCCESS = 'success';
+const ERROR = 'error';
+const INITIALISE = 'initialise';
+const SEARCH_TIMEOUT_INTERVAL = 250;
+const NO_MATCHES_HTML = <p>Sorry, no matches found.</p>;
+const INITIAL_MESSAGE_HTML = (
+  <p>
+    Use the input field above to search for instruments or click on the search
+    icon to close the search.
+  </p>
+);
+
+let latestRequest = null;
+let resultsWindow = null;
+
+const initialSearchState = {
+  isSearching: false,
+  hasErrors: false
+};
+
+const searchReducer = (state, { type, results }) => {
+  switch (type) {
+    case SEARCHING:
+      return {
+        ...state,
+        hasErrors: false,
+        isSearching: true
+      };
+    case SUCCESS:
+      return {
+        ...state,
+        isSearching: false,
+        results
+      };
+    case ERROR:
+      return {
+        ...state,
+        hasErrors: true,
+        isSearching: false
+      };
+    case INITIALISE:
+      return initialSearchState;
+    default:
+      throw new Error();
+  }
+};
+
+const Search = props => {
+  const [searchState, dispatch] = useReducer(searchReducer, initialSearchState);
+  const [parentUuid, setParentUuid] = useState(null);
+  const [query, setQuery] = useState(null);
+  const [windowState] = WindowHooks.useCurrentWindowState();
+  const [debouncedQuery, setDebouncedQuery] = useState(null);
+  const { isSearching, results } = searchState;
+  const searchButtonRef = useRef(null);
+  const searchInputRef = useRef(null);
+
+  const setResultsWindow = newReference => {
+    resultsWindow = newReference;
+  };
+
+  const resetLauncherInputField = () => {
+    if (searchInputRef.current) searchInputRef.current.value = '';
+  };
+
+  const closeResultsWindow = useCallback(() => {
+    if (resultsWindow) {
+      resultsWindow.close();
+      setResultsWindow(null);
+    }
+  }, []);
+
+  const handleOnInputChange = useCallback(
+    event => {
+      setQuery(event.target.value);
+      if (event.target.value !== null && event.target.value.length === 0) {
+        closeResultsWindow();
+      } else if (!resultsWindow)
+        createWindow(
```

Comment:
Needs to match width of search input if possible

Reply:
It seems like this window is also initialised in a way where the search input appears within the window regardless of whether or not the launcher is docked to the top

Reply:
Also, any way to prevent the window from showing up in the taskbar?